packages/ui/components/data-table/filters/utils.ts

```diff
@@ -15,3 +18,111 @@ export const filtersSearchParams = {
 export function useFiltersSearchState() {
   return useQueryStates(filtersSearchParams);
 }
+
+export type FiltersSearchState = ReturnType<typeof useFiltersSearchState>[0];
+export type SetFiltersSearchState = ReturnType<typeof useFiltersSearchState>[1];
+export type ActiveFilter = NonNullable<FiltersSearchState["activeFilters"]>[number];
+
+export const textFilter = (cellValue: string, filterValue: TextFilterValue) => {
```

Comment:
This function is used with tanstack table to filter the existing rows on the client side.